src/proof/drat/drat_proof.h

```diff
@@ -0,0 +1,85 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2022 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * Declares C++ types that represent a DRAT proof.
+ * Defines serialization for these types.
+ *
+ * You can find an introduction to DRAT in the drat-trim paper:
+ *    http://www.cs.utexas.edu/~marijn/publications/drat-trim.pdf
+ */
+
+#include "cvc5_private.h"
+
+#ifndef CVC5__PROOF__DRAT__DRAT_PROOF_H
+#define CVC5__PROOF__DRAT__DRAT_PROOF_H
+
+#include "prop/sat_solver.h"
+#include "prop/sat_solver_types.h"
+
+namespace cvc5::internal {
+namespace proof {
+
+class InvalidDratProofException : public cvc5::internal::Exception
+{
+ public:
+  InvalidDratProofException() : Exception("Invalid DRAT Proof") {}
+
+  InvalidDratProofException(const std::string& msg) : Exception(msg) {}
+
+  InvalidDratProofException(const char* msg) : Exception(msg) {}
+}; /* class InvalidDratProofException */
+
+enum DratInstructionKind
+{
+  ADDITION,
+  DELETION
+};
```

Comment:
Document this. Basically say that a DRAT proof only has two kinds of instructions, which are to add and to delete clauses.